backend/src/main/java/club/devcord/devmarkt/auth/RoleDirective.java

```diff
@@ -52,10 +52,10 @@ public GraphQLFieldDefinition onField(
             .build();
       }
 
-      if (hasRole(user, roleName.getValue()) || hasRole(user, Roles.ADMIN.toString())) {
+      if (user.role() == expectedRole || user.role() == Role.ADMIN) {
```

Comment:
Admins are always permitted to do all (in the moment). The problem with a ordinal number check is that we have to care about the position in the enum, which can easily lead to mistakes. We could introduce a new ordinal number property in the enum, but for the moment I don't see the need for this since we only have 2 roles so far.

Reply:
As long as we just have two roles, this is fine. Once we add/need more roles, we can iterate on this check